src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -1165,8 +1165,11 @@ private void PrepareProjectExecutables()
         }
     }
 
-    private static void EnsureReplicaInstancesAnnotation(IResource resource)
+    private static void EnsureRequiredAnnotations(IResource resource)
     {
+        // Add the default lifecycle commands (start/stop/restart)
+        resource.AddLifeCycleCommands();
```

Comment:
One problem with adding them here is there isn't a way for someone to modify annotations to remove the start/stop/restart commands.

I don't know if anyone would ever want to do that 🤷

Reply:
It's possible to support by checking and no overriding, but I want to fix the gaps we currently have.